File: Leetcode/563-Binary-Tree-Tilt/BinaryTreeTilt.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var findTilt = function(root) {
    var tilt = 0

    var postOrder = function(node) {
        if (!node) {
            return 0
        }
        var left = postOrder(node.left)
        var right = postOrder(node.right)
        tilt += Math.abs(left-right)
        return left+right+node.val
    }
    postOrder(root)
    return tilt
};

File: Leetcode/563-Binary-Tree-Tilt/BinaryTreeTilt.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class BinaryTreeTilt {

    int tilt = 0;

    public int findTilt(TreeNode root) {
        postOrder(root);
        return tilt;
    }

    private int postOrder(TreeNode root) {
        if (root == null) {
            return 0;
        }
        int left = postOrder(root.left);
        int right = postOrder(root.right);
        tilt += Math.abs(left - right);
        return left + right + root.val;
    }
}
